### ISSUE TITLE
useForm type does not work with nested interface types, but objects do

### ISSUE THREAD

**Issue by manuel-rw** [mantinedev/mantine#3141]
### What package has an issue

@mantine/form

### Describe the bug

Hello,

a few hours ago, I tried to implement ``useForm`` in my form.
I use interface types in my application to design the desired structure and define types.
Usually this works fine, but today I came across this bug of ``useForm``.

It seems like, that ``useForm`` works as intended, if you use simple objects inside an interface:
```tsx
interface User {
  name: string;
  properties: {
    enabled: boolean;
    custom?: string;
  }
}
```

But this is not convenient, since I want to extract the properties to their own type:

```tsx
interface User {
  name: string;
  properties: UserProperties;
}

interface UserProperties {
  enabled: boolean;
  custom?: string;
}
```

But this generates the following error:

```
Type '{ enabled: (state: any) => string; custom: (value: any) => string; }' is not assignable to type 'Rule<UserProperties, User>'.
  Object literal may only specify known properties, and 'enabled' does not exist in type 'Rule<UserProperties, User>'.(2322)
```

It seems like the ``Rule<T, T>`` type cannot handle nested interface types.
I wrote two definitions of the same structure in the reproduced example below. Switch between them to see it happening.

I am happy to provide more information if necessary.

JFYI: @Meierschlumpf 

### What version of @mantine/hooks page do you have in package.json?

5.9.2

### If possible, please include a link to a codesandbox with the reproduced problem

https://stackblitz.com/edit/react-ts-2gxdzc?file=App.tsx

### Do you know how to fix the issue

No

### Are you willing to participate in fixing this issue and create a pull request with the fix

Yes

**Comment by rtivital** [comment]
@manuel-rw why did you close the issue?

**Comment by manuel-rw** [comment]
Sorry, pressed by mistake...

**Comment by rtivital** [comment]
Fixed in 5.9.3
